frontend/test/validate-e2e-test-files.js

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env node
+const { execSync } = require("child_process");
+
+const path = require("path");
+const glob = require("glob");
+const chalk = require("chalk");
+
+const E2E_FILE_EXTENSION = ".cy.spec.js";
+const E2E_HOME = "frontend/test/metabase/scenarios/";
+
+init();
+
+function validateE2EFileNames(files) {
+  if (!files || !Array.isArray(files)) {
+    return;
+  }
+
+  const invalidFileNames = files.filter(fullPath => {
+    return !path.basename(fullPath).endsWith(E2E_FILE_EXTENSION);
+  });
+
+  printFeedback(invalidFileNames);
+}
+
+/**
+ * @param {Array<string>} [files] Potential payload if used with `lint-staged`
+ */
+function validateStagedFiles(files) {
+  const stagedFiles = getStagedFiles();
+  validateE2EFileNames(files || stagedFiles);
+}
+
+function validateAllFiles() {
+  const allE2EFiles = getAllE2EFiles();
+  validateE2EFileNames(allE2EFiles);
+}
+
+function getAllE2EFiles() {
+  // Will match all files in the scenarios dir, except the ones in helpers and shared directories
+  const PATTERN = `${E2E_HOME}*/{*.js,!(helpers|shared)/*.js}`;
+
+  return glob.sync(PATTERN);
+}
+
+function getStagedFiles() {
+  const stagedFiles = execSync("git diff HEAD --name-only --diff-filter=d", {
+    encoding: "utf8",
+  });
+  const relevantStagedFiles = stagedFiles.includes(E2E_HOME);
+
+  if (relevantStagedFiles) {
+    return stagedFiles.split("\n").filter(isE2ETestFile);
```

Comment:
Not sure if this should be
```suggestion
    return relevantStagedFiles.split("\n").filter(isE2ETestFile);
```

Otherwise, it will look odd that `relevantStagedFiles` is only used as a condition.

Reply:
`String.prototype.includes()` returns a boolean value. It was meant to be used only as a condition.

Reply:
Cool, I misread that entirely. It's just that `relevantStagedFiles` sounds more like a list of files if you even compare that to `stagedFiles` (try saying both names).

I'd suggest changing this to follow our boolean naming convention maybe `hasRelevantStagedFiles`?

Reply:
Thanks.
Will do that.